src/webgpu/api/validation/attachment_compatibility.spec.ts

```diff
@@ -350,44 +289,50 @@ Test that the sample count in render passes or bundles match the pipeline sample
   .params(u =>
     u
       .combine('encoderType', ['render pass', 'render bundle'] as const)
+      .combine('attachmentType', ['color', 'depthstencil'] as const)
       .beginSubcases()
       .combine('encoderSampleCount', kTextureSampleCounts)
       .combine('pipelineSampleCount', kTextureSampleCounts)
   )
   .fn(t => {
-    const { encoderType, encoderSampleCount, pipelineSampleCount } = t.params;
-
-    // For color texture
-    const pipelineWithoutDepthStencil = t.createRenderPipeline(
-      [{ format: 'rgba8unorm' }],
-      undefined,
-      pipelineSampleCount
-    );
-
-    const {
-      encoder: encoderWithoutDepthStencil,
-      finish: finishWithoutDepthStencil,
-    } = t.createPassOrBundleEncoder(encoderType, ['rgba8unorm'], undefined, encoderSampleCount);
-    encoderWithoutDepthStencil.setPipeline(pipelineWithoutDepthStencil);
-
-    t.expectValidationError(() => {
-      t.queue.submit([finishWithoutDepthStencil()]);
-    }, encoderSampleCount !== pipelineSampleCount);
-
-    // For DepthStencil texture
-    const pipelineWithDepthStencilOnly = t.createRenderPipeline(
-      [],
-      { format: 'depth24plus-stencil8' },
-      pipelineSampleCount
-    );
-
-    const {
-      encoder: encoderWithDepthStencilOnly,
-      finish: finishWithDepthStencilOnly,
-    } = t.createPassOrBundleEncoder(encoderType, [], 'depth24plus-stencil8', encoderSampleCount);
-    encoderWithDepthStencilOnly.setPipeline(pipelineWithDepthStencilOnly);
-
-    t.expectValidationError(() => {
-      t.queue.submit([finishWithDepthStencilOnly()]);
-    }, encoderSampleCount !== pipelineSampleCount);
+    const { encoderType, attachmentType, encoderSampleCount, pipelineSampleCount } = t.params;
+
+    switch (attachmentType) {
```

Comment:
nit(since the original code was just chaining the expects): can't we replace the switch with something like
```ts
  let colorFormats = attachmentType === "color" ? ["rgba8unorm"] : [];
  let depthStencilFormat =
    attachmentType === "depthstencil" ? "depth24plus-stencil8" : undefined;

  const pipeline = t.createRenderPipeline(
    colorFormats.map((format) => ({ format })),
    depthStencilFormat === undefined ? { format: depthStencilFormat } : undefined,
    pipelineSampleCount
  );

  const { encoder, validateFinishAndSubmit } = t.createEncoder(encoderType, {
    attachmentInfo: {
      colorFormats,
      depthStencilFormat,
      sampleCount: encoderSampleCount,
    },
  });

  encoder.setPipeline(pipeline);
  validateFinishAndSubmit(encoderSampleCount === pipelineSampleCount, true);
```

Reply:
Good call!